src/workerd/io/worker-fs.h

```diff
@@ -0,0 +1,203 @@
+ public:
+  FsMap() = default;
+  KJ_DISALLOW_COPY_AND_MOVE(FsMap);
+
+#define V(name, _)                                                                                 \
+  static const jsg::Url kDefault##name##Path;                                                      \
+  const jsg::Url& get##name##Root() const {                                                        \
+    return maybe##name##Root.orDefault(kDefault##name##Path);                                      \
+  }                                                                                                \
+  const kj::Path get##name##Path() const {                                                         \
+    auto path = kj::str(get##name##Root().getPathname().slice(1));                                 \
+    return kj::Path::parse(path);                                                                  \
+  }                                                                                                \
+  void set##name##Root(jsg::Url url) {                                                             \
+    KJ_REQUIRE(url.getProtocol() == "file:"_kj, "url must be a file URL");                         \
+    maybe##name##Root = kj::mv(url);                                                               \
+  }                                                                                                \
+  void set##name##Root(kj::StringPtr path) {                                                       \
+    auto url = KJ_REQUIRE_NONNULL(jsg::Url::tryParse(path, "file:///"_kj), "invalid path");        \
+    set##name##Root(kj::mv(url));                                                                  \
+  }
+
+  KNOWN_VFS_ROOTS(V)
+
+#undef V
+
+ private:
+#define V(name, _) kj::Maybe<jsg::Url> maybe##name##Root;
+  KNOWN_VFS_ROOTS(V)
+#undef V
+};
+
+// An RAII object that stores the temporary directory items.
+// Instances can either live on the stack (in which case they
+// will be set in a thread-local and hasCurrent() will return
+// true, or they can be created on the heap and held (e.g. in
+// IoContext).
+//
+// By default, all file instances are created using the UnixEpochClock,
+// which sets all timestamps to the Unix epoch. When creating a new
+// TempDirStoreScope, you can pass a different clock to set the
+// timestamps. This is particularly important when setting timestamp
+// for files created under an IoContext, in which case the we want
+// to make sure the times are set to the correct time as returned by
+// IoContext::now(), correctly reflecting time in for that request.
+class TempDirStoreScope final {
+ public:
+  static bool hasCurrent();
+  static TempDirStoreScope& current();
+  TempDirStoreScope(kj::Maybe<kj::Badge<TempDirStoreScope>> guard = kj::none);
+  KJ_DISALLOW_COPY_AND_MOVE(TempDirStoreScope);
+  ~TempDirStoreScope() noexcept(false);
+
+  static kj::Own<TempDirStoreScope> create();
+
+  jsg::fs::Directory& getDirectory() {
```

Comment:
```suggestion
  const jsg::fs::Directory& getDirectory() {
```

Reply:
The directory returned is not `const` tho. This wouldn't make sense.